services/src/handlers/plots.rs

```diff
@@ -179,7 +181,10 @@ async fn get_plot_handler<C: Context>(
         });
     };
 
-    let processor = initialized.query_processor().context(error::Operator)?;
+    let processor = execution_context
+        .create_plot_query_processor(initialized)
+        .await
+        .context(error::Operator)?;
```

Comment:
1. What happens if we call it like before?
2. Why is it wrapped here, but all other methods just take a reference to the execution context?

Reply:
1. If you create the processor from the InitializedOperator, like before, then no quota check is performed. This here is basically the point where, in Pro, the quota is checked, and the execution prevented if it is less than zero. In free, no check is performed, because there is no quota.

2. I did it this way to implement a different behavior, depending on the context being Pro or Free. That#s why `create_plot_query_processor` is a trait method with different implementations depending on the type of context.

I am sure there are plenty other ways to implement the quota check. E.g. duplicating the WMS handler code in Pro, and including a check there. But I found my solution to be the easiest, least intrusive and most effective.